python/tests/integration/helpers.py

```
@@ -24,6 +26,141 @@
 DEFAULT_TIMEOUT = 20.0
 
 
+def child_pids(parent_pid: int) -> set[int]:
+    """Return the PIDs of a process's direct children (POSIX only).
+
+    Used to observe worker recycling: the ``runbolt`` supervisor forks one
+    worker per slot, so a changing child-PID set is proof of recycles.
+    """
+    result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)